lib/shp-reader.js

```diff
@@ -82,7 +90,14 @@ ShpReader.prototype.readRecord = function (callback) {
       }
 
       // shapeType is good, parse contentBuffer
-      const content = new shapeTypeParsers[SHAPE_TYPES[shapeType]](contentBuffer)
+      let content
+      if (shapeTypeParsers.hasOwnProperty(SHAPE_TYPES[shapeType])) {
+        content = new shapeTypeParsers[SHAPE_TYPES[shapeType]](contentBuffer)
+      } else {
```

Comment:
Shouldn't this throw/emit an error eventually instead of just silently passing through?